midend/predication.cpp

```diff
@@ -38,6 +39,8 @@ static cstring lvalue_name(const IR::Expression *exp) {
     return cstring();
 }
 
+// Remove all empty statements from the code.
+// For each assignmentStatement in the source code an emptyStatement is generated.
```

Comment:
This comments seems wrong.
Also, please use /// comments when describing these functions.
Moreover, this code could be much simpler using a visitor: just return nullptr for an EmptyStatement. In fact, we have some other passes which do this already.